pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1312,23 +1327,39 @@ func (qjm *XController) ScheduleNext() {
 						} else { // successful add to eventQueue, remove from qjqueue
 							qjm.qjqueue.Delete(qj)
 							forwarded = true
-							klog.V(3).Infof("[ScheduleNext] %s Delay=%.6f seconds eventQueue.Add_afterHeadOfLine activeQ=%t, Unsched=%t &aw=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(qj.Status.ControllerFirstTimestamp.Time).Seconds(), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
+							klog.V(4).Infof("[ScheduleNext] %s Delay=%.6f seconds eventQueue.Add_afterHeadOfLine activeQ=%t, Unsched=%t &aw=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(qj.Status.ControllerFirstTimestamp.Time).Seconds(), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
 						}
 					} //updateEtcd
 				} //fits
 			} else { // Not enough free resources to dispatch HOL
 				dispatchFailedMessage = "Insufficient resources to dispatch AppWrapper."
-				klog.V(3).Infof("[ScheduleNext] HOL Blocking by %s for %s activeQ=%t Unsched=%t &qj=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(HOLStartTime), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
+				klog.V(4).Infof("[ScheduleNext] HOL Blocking by %s for %s activeQ=%t Unsched=%t &qj=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(HOLStartTime), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
 			}
-			// stop trying to dispatch after HeadOfLineHoldingTime
-			if forwarded || time.Now().After(HOLStartTime.Add(time.Duration(qjm.serverOption.HeadOfLineHoldingTime)*time.Second)) {
+			schedulingTimeExpired := time.Now().After(HOLStartTime.Add(time.Duration(qjm.serverOption.HeadOfLineHoldingTime) * time.Second))
+			if forwarded {
+				break
+			} else if schedulingTimeExpired {
+				// stop trying to dispatch after HeadOfLineHoldingTime
+				// release quota if allocated
+				if qjm.quotaManager != nil && quotaFits {
```

Comment:
@asm582, @KPostOffice, @metalcycling in your testing have you seen this condition happen ?

Reply:
@asm582 the HeadofLineHoldingTime is a command line option, defined at:
[fs.IntVar(&s.HeadOfLineHoldingTime, "headoflineholdingtime", s.HeadOfLineHoldingTime, "Number of seconds a job can stay at the Head Of Line without being bumped.  Default is 0.")](https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/b32182ce790f31affd79246cd635e3f24e4c899f/cmd/kar-controllers/app/options/options.go#L80)

AFAIK, the helm chart does not define an override